### PR TITLE
add per sample option for affine transforms

### PR DESCRIPTION
**Pull request by mibaumgartner**
## Short Description
* per sample option for affine transform

### PR Implementer
This is a small checklist for the implementation details of this PR.
If you submit a PR, please look at these points (don't worry about the `RisingTeam`
and `Reviewer` workflows, the only purpose of those is to have a compact view of
the steps)

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/PhoenixDL/rising/blob/master/CONTRIBUTING.md).

- [x] Implementation
- [x] Docstrings & Typing
- [x] Check `__all__` sections and `__init__`
- [x] Unittests (look at the line coverage of your tests, the goal is 100%!)
- [x] Update notebooks & documentation if necessary
- [x] Pass all tests
- [x] Add the checksum of the last implementation commit to the Changelog

### RisingTeam
<details>
  <summary>RisingTeam workflow</summary>

  - [ ] Add pull request to project (optionally delete corresponding project note)
  - [ ] Assign correct label (if you don't have permission to do this, someone will do it for you. 
      Please make sure to communicate the current status of the pr.)
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)
</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>
  
  - [ ] Do all tests pass? (Unittests, NotebookTests, Documentation)
  - [ ] Does the implementation follow `rising` design conventions?
  - [ ] Are the tests useful? (!!!) Are additional tests needed? 
        Can you think of critical points which should be covered in an additional test?
  - [ ] Optional: Check coverage locally / Check tests locally if GPU is necessary to execute
</details>

### CODE REVIEW

File: rising/transforms/affine.py [hunk 1]
```diff
@@ -37,6 +37,7 @@ def __init__(self,
                  padding_mode: str = 'zeros',
                  align_corners: bool = False,
                  reverse_order: bool = False,
+                 per_sample: bool = True,
```
Comment (justusschock) [inline rising/transforms/affine.py:40]: just wondering, if we should let this default to false for now to be consistent to the other trafos and change this in the future.

Comment (mibaumgartner) [reply rising/transforms/affine.py:40]: I was also thinking about that. I think we will need to decide on a per transform basis. 
E.g. in this case there is no drawback to have it on
With our current mirror implementation on the other we will need a loop to do this whil won't be the best performance? So I would leave it false there.
In general we will need to introduce automatic stacking of affine transforms und refactor some of our spatial transforms to affine ones to get max performance for the per sample case

Comment (justusschock) [reply rising/transforms/affine.py:40]: > So I would leave it false there.

You mean leave it true? Or change it to false?

Comment (mibaumgartner) [reply rising/transforms/affine.py:40]: I mean leave it true 🙈

**Pull request merged**